### PR TITLE
Added processor details handler

### PR DESCRIPTION
**Pull request by mcgordonite**
CPRJ projects target a specific processor on a device. The debugger needs to know which processor to connect to when debugging a program. This new handler will enable this.

The processor name doesn't fit nicely into the existing interface, possibly due to incompatibilities between the Mbed and CMSIS approaches. Options considered:
- Program namespace. This makes sense in the CPRJ world, where the procesor name is specific to a program. However, this doesn't make sense in Mbed, or a future CMSIS project structure supporting multi-core.
- Target namespace. The processor name could be added to the Target interface. However, Targets are currently synonymous with development boards, but there is a one-to-many mapping between boards and processors.
- Debug namespace. This makes sense for our current use case of supporting debug, but in future we might want to show the processor in association with the program.

Picking debug for now, since it solves the current problem. We may need to revisit this in future.

Inside the debug interface, I chose to create a new method since it didn't clearly fit into an existing one:
- It doesn't make sense in `cmsisFilesHandler`, since it doesn't depend on the device.
- It doesn't make sense in `debugDetailsHandler`, since it doesn't depend on the target.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -311,6 +311,19 @@ declare module '@arm-debug/mbs-plugin' {
             path: string;          // Path to extracted file
         }
 
+        export interface ProcessorDetails {
```
Comment (thegecko) [inline index.d.ts:314]: Can we make this a more generic name? I could imagine other details (family, device name, etc.) could be added to this longer term.

Shame we can't use `debugDetails`, what about `cmsisDetails`?

**Review by thegecko — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 1 above)
Comment (mcgordonite) [reply index.d.ts:314]: I like the idea of returning the DAP config from the plugin instead, but that's for later.

Could be `TargetDetails`, or `DebugTargetDetails` (but it's already in the debug namespace).

Comment (mcgordonite) [reply index.d.ts:314]: Actually, "target" is already overloaded in this interface, happy to go with `CmsisDetails`.

**Review by thegecko — APPROVED** [review]

**Comment by thegecko** [comment]
Approved, probably need a version bump

**Pull request merged**